docs/frontend-system/building-plugins/04-built-in-data-refs.md

```diff
@@ -0,0 +1,103 @@
+---
+id: built-in-data-refs
+title: Built-in data refs
+sidebar_label: Built-in data refs
+# prettier-ignore
+description: Configuring or overriding built-in extension data references
+---
+
+> **NOTE: The new frontend system is in alpha and is only supported by a small number of plugins.**
+
+To have a better understanding of extension data references please read [extension data section the extensions architecture documentation](../architecture/03-extensions.md#extension-data) first.
+
+## Built-in extension data references
+
+Data references help to define the input & output of an extension. A data ref is uniquely identified through there `id`. Through the data ref strong typing is enforced for the input/output of the extension.
+
+### Core Extension Data
+
+Commonly used `ExtensionDataRef`s for extensions.
+
+| namespace |     name     |     type      |         id          |
+| :-------: | :----------: | :-----------: | :-----------------: |
+|     -     | reactElement | `JSX.Element` | `core.reactElement` |
+|     -     |  routePath   |   `string`    | `core.routing.path` |
+|     -     |   routeRef   |  `RouteRef`   | `core.routing.ref`  |
+
+### Core API Data
+
+|     namespace      |      name      |      type       |         id         |
+| :----------------: | :------------: | :-------------: | :----------------: |
+| createApiExtension | factoryDataRef | `AnyApiFactory` | `core.api.factory` |
```

Comment:
Hm these are not namespaces. The namespace typically gets set to the plugin ID later down the line; the factory functions create an ExtensionDefinition which later becomes an Extension during instantiation.

Maybe it's better to point to the extension creator being `createApiExtension`, and the ref being `createApiExtension.factoryDataRef`.

A couple words on each of these explaining what they do would be nice. Not much required, just a few words stolen from e.g. the architecture section and then linking to that exact section too.

For these refs that you normally do not consume yourself, it can be useful to say so - something like "This extension data ref ref is normally never used directly in plugin code. It's only produced by the extension creator to be consumed by the app framework." or similar

Reply:
Yeah, I struggled a bit with the naming here as they are technically TypeScript namespaces, but it is a bit confusing in the context of Extensions & ID's. 

Ok - good point, will do that!:)